packages/ui/src/helper.ts

```diff
@@ -463,6 +463,9 @@ const handleTypeChange = (
       (schema as any)[key] = propPanel?.defaultSchema[key];
     }
   });
+  if (schema.readOnly) {
```

Comment:
[option]  
If we can use  `options.requiredByDefault`,  `schema.required` should be determine by this option.

Reply:
this is handled on [line 174 of `Designer/index.tsx` ](https://github.com/pdfme/pdfme/blob/pw-required-fields/packages/ui/src/components/Designer/index.tsx#L175)- when a new field is added it is set here. A readOnly schema will always not be required.

Reply:
Ah, Okay👍